jbehave-support-core/src/main/java/org/jbehavesupport/core/report/extension/JmsXmlReporterExtension.java

```diff
@@ -42,44 +42,65 @@ public void print(Writer writer, ReportContext reportContext) {
     }
 
     private void printJmsMessage(Writer writer, Message message) {
+        printBegin(writer, MESSAGE_TAG);
+
+        String cid;
         try {
-            printBegin(writer, MESSAGE_TAG);
-            printBegin(writer, CID_TAG);
-            printCData(writer, message.getJMSCorrelationID());
-            printEnd(writer, CID_TAG);
-
-            printBegin(writer, ANSWERS_TAG);
-            Arrays.stream(headers).forEach(header -> {
-                printBegin(writer, ROW_TAG);
-                printBegin(writer, COLUMN_TAG);
-                printCData(writer, header);
-                printEnd(writer, COLUMN_TAG);
-
-                printBegin(writer, COLUMN_TAG);
-                try {
-                    printCData(writer, message.getStringProperty(header));
-                } catch (JMSException e) {
-                    log.error("Unable to get the property: {}", header, e);
-                    printCData(writer, "Unable to get property: " + header);
-                }
-                printEnd(writer, COLUMN_TAG);
-
-                printEnd(writer, ROW_TAG);
-            });
-
-            printBegin(writer, STRING_TAG);
-            if (message instanceof TextMessage) {
-                TextMessage textMessage = (TextMessage) message;
+            cid = message.getJMSCorrelationID();
+        } catch (JMSException e) {
+            cid = "00000";
```

Comment:
How about not printing it at all, or printing some message along the line in error instead of five zeros? (Plus what if you would get rid of the cid variable and just use printCID straight away - might be more readable?)

Reply:
Some error message would be useful. However, the variable is needed, because CID is used as a header for the whole message and if it failed to print, the report would be hard to navigate